server/src/main/java/org/apache/cloudstack/backup/BackupManagerImpl.java

```diff
@@ -671,6 +667,14 @@ public boolean deleteBackup(final Long backupId) {
         if (offering == null) {
             throw new CloudRuntimeException("VM backup offering ID " + vm.getBackupOfferingId() + " does not exist");
         }
+        List<Backup> backupsForVm = backupDao.listByVmId(vm.getDataCenterId(), vmId);
+        if (CollectionUtils.isNotEmpty(backupsForVm)) {
+            backupsForVm = backupsForVm.stream().filter(vmBackup -> vmBackup.getId() != backupId).collect(Collectors.toList());
+            if (backupsForVm.size() <= 0 && vm.getRemoved() != null) {
+                removeVMFromBackupOffering(vmId, true);
```

Comment:
This is also because, there may be a backup policy. Please check that also - potentially can cause a regression.

Reply:
deletion of backup schedule is handled as part of removeVMFromBackupOffering()

Reply:
My worry is deletion of single last backup may remove it? And if backup policy could have an related regression. For example on expunging a VM should we expunge a backup policy? Or not?

Reply:
This logic basically deletes the backup schedule and unassigns the backup offering when the vm being expunged has NO backups